### PR TITLE
fix open_pdfm_filter_init double promotion

### PR DESCRIPTION
**Pull request by XenuIsWatching**
floating point constants were defined as doubles. change them to single-precision point to prevent double promotion of the whole thing.

### CODE REVIEW

**Comment by cfriedt** [comment]
@abostm @frastm

**Review by cfriedt — APPROVED** [review]
Erwan is maintainer here, so he should approve before it gets merged, but this one is trivial, so I will stamp 👍

**Review by keith-packard — APPROVED** [review]

**Comment by erwango** [comment]
@avisconti would you mind having a look ? Looks fine to me but I'd like to be sure this will be reported in initial library.

@XenuIsWatching Would you mind adding a note in README (similar to what is done [here](https://github.com/zephyrproject-rtos/hal_stm32/blob/main/stm32cube/stm32l4xx/README#L37) so that this doesn't get overwritten in case the library is updated ?

**Comment by XenuIsWatching** [comment]
> @avisconti would you mind having a look ? Looks fine to me but I'd like to be sure this will be reported in initial library.
> 
> @XenuIsWatching Would you mind adding a note in README (similar to what is done [here](https://github.com/zephyrproject-rtos/hal_stm32/blob/main/stm32cube/stm32l4xx/README#L37) so that this doesn't get overwritten in case the library is updated ?

done

**Review by erwango — APPROVED** [review]

**Pull request merged**
